server/datastore/mysql/scheduled_queries.go

```diff
@@ -145,3 +199,35 @@ func (d *Datastore) ScheduledQuery(id uint) (*fleet.ScheduledQuery, error) {
 
 	return sq, nil
 }
+
+func (d *Datastore) ScheduledQueryByQueryID(id uint) (*fleet.ScheduledQuery, error) {
+	query := `
+		SELECT
+			sq.id,
+			sq.created_at,
+			sq.updated_at,
+			sq.pack_id,
+			sq.interval,
+			sq.snapshot,
+			sq.removed,
+			sq.platform,
+			sq.version,
+			sq.shard,
+			sq.query_name,
+			sq.description,
+			sq.denylist,
+			q.query,
+			q.name,
+			q.id AS query_id
+		FROM scheduled_queries sq
+		JOIN queries q
+		ON sq.query_name = q.name
+		WHERE q.id = ?
```

Comment:
I think this assumes that each query would be scheduled in at most one pack (including global)? A query can actually be scheduled multiple times within the same pack and/or global pack.

Reply:
Ah, yes, my understanding was incorrect. I misread the indexes.